protocol/client/messenger_v2.go

```diff
@@ -157,19 +162,55 @@ func (m *MessengerV2) Request(ctx context.Context, c Contact, options protocol.R
 	return m.proto.Request(ctx, options)
 }
 
+func (m *MessengerV2) requestHistories(ctx context.Context, histories []History, opts protocol.RequestOptions) error {
+	log.Printf("[messenger::RequestAll] requesting messages for chats %+v: from %d to %d\n", opts.Chats, opts.From, opts.To)
+	start := time.Now()
+	err := m.proto.Request(ctx, opts)
+	if err != nil {
+		return err
+	}
+	log.Printf("[messenger::RequestAll] requesting message for chats %+v finished. took %v\n", opts.Chats, time.Since(start))
+	for i := range histories {
+		histories[i].Synced = opts.To
+	}
+	err = m.db.UpdateHistories(histories)
+	return err
+}
+
 func (m *MessengerV2) RequestAll(ctx context.Context, newest bool) error {
-	contacts, err := m.db.Contacts()
+	// FIXME(dshulyak) if newest is false request 24h?
```

Comment:
The idea with `newest` was that it's `false` if one scrolls up. It means the older and older messages should be requested and displayed.

Reply:
we need then to store earliest timestamp for every contact, right?

Reply:
Yeah to avoid sending requests to mailserver. It can be a separate issue.